File: src/main/java/problems/arrays/minimumsizesubarraysum/MinimumSizeSubarraySum.java
package problems.arrays.minimumsizesubarraysum;

public class MinimumSizeSubarraySum {


    public static void main(String[] args){


        int[] nums = {2,3,1,1,4,3};
        int target = 7;

        System.out.print("minimum length -> "+minSubarraySum(target,nums));
    }

    public static int minSubarraySum(int target, int[] nums){

        int minWindowLen = Integer.MAX_VALUE;
        int currentSum = 0;

        int low = 0;
        int high = 0;

        while (high < nums.length){

            currentSum += nums[high];
            high++;

            while (currentSum >= target){

                int currentWindowLen = high -low;

                minWindowLen = Math.min(currentWindowLen, minWindowLen);

                currentSum -= nums[low];
                low++;
            }
        }

        return minWindowLen == Integer.MAX_VALUE ? 0 : minWindowLen;
    }
}

File: src/main/java/problems/arrays/minimumsizesubarraysum/MinimumSizeSubArraySum.kt
package problems.arrays.minimumsizesubarraysum


fun main(){

    var nums = intArrayOf(2,3,1,2,4,3)
    var target = 7

    println("result -> ${minSubarraySum(target,nums)}")

}

fun minSubarraySum(target: Int, nums: IntArray):Int{


    var minWindowLen = Int.MAX_VALUE
    var currentSum = 0;

    var low = 0;
    var high = 0;

    while (high < nums.size){

        currentSum += nums[high]
        high++

        while (currentSum >= target){

            var currentWindowLen = high - low

            minWindowLen = kotlin.math.min(currentWindowLen, minWindowLen)
            currentSum -= nums[low];
            low++
        }
    }

    return if (minWindowLen == Int.MAX_VALUE) 0 else minWindowLen
}